pkg/visitor/fetch_test.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2023 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package visitor
+
+import (
+	"context"
+	"testing"
+
+	"github.com/apigee/registry/pkg/connection"
+	"github.com/apigee/registry/rpc"
+	"github.com/apigee/registry/server/registry/names"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func TestFetch(t *testing.T) {
+	projectID := "fetch-test"
+	project := names.Project{ProjectID: projectID}
+	parent := project.String() + "/locations/global"
+
+	specContents := "hello"
+	artifactContents := "hello"
+
+	ctx := context.Background()
+	adminClient, err := connection.NewAdminClient(ctx)
```

Comment:
Do we want to start using `grpctest.SetupRegistry()`?

Reply:
Yes, let's do some dedicated PRs for that change. I would have used it here but didn't feel that I knew the pattern well enough yet.